### PR TITLE
Feature/serial episodic

### PR DESCRIPTION
**Pull request by drmikecrowe**
Add support for <itunes:type> serial/episodic flag.  Details are [here](https://help.apple.com/itc/podcasts_connect/#/itcb54353390)

No change  to current operation if not provided

### CODE REVIEW

**Comment by drmikecrowe** [comment]
Good feedback, let me work on that

File: podcastparser.py [hunk 1]
```diff
@@ -686,6 +686,8 @@ def set_base(self, base):
         self.base = base
 
     def set_podcast_attr(self, key, value):
+        if key == 'type' and value not in ['episodic', 'serial']:
```
Comment (thp) [inline podcastparser.py:689]: This can be a tuple: `... not in ('episodic', 'serial')`

It shouldn't matter performance-wise in recent Python 3 versions (see [this Stackoverflow question](https://stackoverflow.com/questions/25368337/tuple-or-list-when-using-in-in-an-if-clause)) but IMHO is "clearer".

File: podcastparser.py [hunk 2]
```diff
@@ -661,7 +662,6 @@ def parse_pubdate(text):
 class FeedParseError(sax.SAXParseException, ValueError):
     """
     Exception raised when asked to parse an invalid feed
-
```
Comment (thp) [inline podcastparser.py:664]: Unrelated whitespace change. Maybe remove.

File: podcastparser.py [hunk 3]
```diff
@@ -580,7 +581,6 @@ def parse_pubdate(text):
         except(OverflowError,ValueError):
             logger.warning('bad pubdate %s is before epoch or after end of time (2038)',parsed)
             return 0
-
```
Comment (thp) [inline podcastparser.py:583]: Unrelated whitespace change. Maybe remove.

File: podcastparser.py (hunk 1 above)
Comment (thp) [inline podcastparser.py:689]: Instead of doing this check here, it might also have been possible (cleaner?) to subclass `PodcastAttr` (e.g. `PodcastAttrOneOf('type', squash_whitespace, ('episodic', 'serial'))`) and deal with the special case there instead of here. I mean this definition above:

        'rss/channel/itunes:type': PodcastAttr('type', squash_whitespace),

Comment (drmikecrowe) [reply podcastparser.py]: I took a slightly different approach that seemed OK -- let me know if you want me to go the full `PodcastAttrOneOf`

**Comment by drmikecrowe** [comment]
OK, *most* of your changes.  I can't seem to fix those whitespaces easily (vscode is trimming trailing whitespaces and I can't seem to turn it off).

How strongly do you feel about it?  I can manually edit it in vim or something and add those spaces in those lines if you feel strongly about it.
